### PR TITLE
Add chat publicity trigger

### PR DESCRIPTION
**Pull request by sheriffjimoh**
https://trello.com/c/H0bVCwGj

## Summary by Sourcery

Add a new feature to manage thread visibility with a CustomSwitch component, allowing users to toggle threads between public and private. Refactor the ThreadPopup component to integrate this new switch. Clean up the code by removing unnecessary console.log statements and update a comment link in the TypeScript configuration.

New Features:
- Introduce a CustomSwitch component to toggle thread visibility between public and private states.

Enhancements:
- Refactor the ThreadPopup component to include the new CustomSwitch for managing thread visibility.

Chores:
- Remove console.log statements from various components to clean up the codebase.
- Update a comment in the next-env.d.ts file to reflect the latest Next.js documentation link.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `ThreadPublicitySwitch` component for managing thread visibility.
	- Enhanced `ThreadPopup` layout with a new toggle switch feature and improved organization.
	- Integrated a new hook `useThreadVisibility` for managing thread visibility state.
- **Bug Fixes**
	- Removed debug `console.log` statements from various components for cleaner code.
- **Chores**
	- Added a new dependency for the AI SDK for Anthropic to enhance AI capabilities.
	- Updated the JWT secret in the `docker-compose.yml` for improved authentication security.

### CODE REVIEW

**Comment by AndlerRL** [comment]
@sheriffjimoh — After checking the preview, it seems that the UI component works but is failing to update the database. See:

DEMO


https://github.com/user-attachments/assets/2b2d56d0-0f9f-4448-b004-efdc616c01d1

> Check the table permissions to update, that might be the issue.
> Please check conflicts with develop, there has been recent folder structure updates on the component folder. 👀

File: apps/masterbots.ai/components/ui/custom-switch.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import * as Switch from '@radix-ui/react-switch';
+import { useState } from 'react';
+import { Eye, EyeOff } from 'lucide-react'; // Importing the icons from lucide-react
+import { UpdateThreadVisibility } from '@/services/hasura';
+import { useSession } from 'next-auth/react'
+
+export  function CustomSwitch({ threadId }: { threadId: string }) {
```
[the remark this answers is not in the document]
Comment (AndlerRL) [reply apps/masterbots.ai/components/ui/custom-switch.tsx:7]: @sheriffjimoh -- please apply this improvement given by @sourcery-ai. It is always good to have the components to deal with the markup, styling and styling conditions (if any) while the functionality in custom hooks.

File: apps/masterbots.ai/components/ui/custom-switch.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { Eye, EyeOff } from 'lucide-react'; // Importing the icons from lucide-react
+import { UpdateThreadVisibility } from '@/services/hasura';
+import { useSession } from 'next-auth/react'
+
+export  function CustomSwitch({ threadId }: { threadId: string }) {   
+  const [isChecked, setIsChecked] = useState(false);
+  const session = useSession()
+  const jwt = session?.data?.user?.hasuraJwt
+
+  const handleSwitchChange = async (checked: boolean) => {
```
[the remark this answers is not in the document]
Comment (AndlerRL) [reply apps/masterbots.ai/components/ui/custom-switch.tsx:12]: @sheriffjimoh -- good improvement here :eyes: don't let behind any async/await call without controlling errors.

File: apps/masterbots.ai/components/ui/custom-switch.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    >
+      <Switch.Thumb
+        className={`w-[24px] h-[24px] bg-gray-100 dark:bg-gray-800 rounded-full transform transition-transform duration-100 ease-in-out ${isChecked ? 'translate-x-[24px]' : ''}`}
+      />
+      <div className="absolute left-0 right-0 flex justify-between items-center w-full h-full px-[4px]">
+        <Eye className={`w-[16px] h-[16px] transition-opacity duration-300 ${isChecked ? 'text-gray-400' : 'text-gray-200'}`} />
+        <EyeOff className={`w-[16px] h-[16px]  transition-opacity duration-300 ${isChecked ? 'text-gray-200' : 'text-gray-400'}`} />
+      </div>
+    </Switch.Root>
+  );
```
[the remark this answers is not in the document]
Comment (AndlerRL) [reply apps/masterbots.ai/components/ui/custom-switch.tsx:31]: @sheriffjimoh -- can we add the `title` html attribute to show this information? e.g.: "Make private" | "Make public".

And always use `cn` from our utils to do conditional className assignation :eyes:

File: apps/masterbots.ai/components/ui/custom-switch.tsx
Comment (AndlerRL) [inline apps/masterbots.ai/components/ui/custom-switch.tsx:1]: Can we rename this file to `thread-publicity-switch`  and move what it to the `routes/thread` folder?

File: apps/masterbots.ai/services/hasura/hasura.service.ts [hunk 4]
```diff
@@ -554,3 +554,23 @@ export async function getUsers() {
   })
   return user as User[]
 }
+
+export async function UpdateThreadVisibility({
```
[the remark this answers is not in the document]
Comment (AndlerRL) [reply apps/masterbots.ai/services/hasura/hasura.service.ts:558]: @sheriffjimoh -- Please control async/await calls :eyes:

File: docker-compose.yml [hunk 5]
```diff
@@ -38,7 +38,7 @@ services:
       HASURA_GRAPHQL_UNAUTHORIZED_ROLE: "anonymous"
       HASURA_GRAPHQL_EXPERIMENTAL_FEATURES: naming_convention
       HASURA_GRAPHQL_DEFAULT_NAMING_CONVENTION: graphql-default
-      HASURA_GRAPHQL_JWT_SECRET: '{"type":"HS256", "key":"xx52fa850c02dc222631cca898ed1485821a70912a6e3649c49076912daa3b62182ba013315915d64f402ddfbb8b58eb5bd11ba225136a6af45bbae07ca872f4"}'
+      HASURA_GRAPHQL_JWT_SECRET: '{"type":"HS256", "key":"cea97d37d22705812fd5244b99a7fa7d4f356235c44c8be30ae9699979f46bff"}'
```
Comment (AndlerRL) [inline docker-compose.yml:41]: Can we restore to the previous value? We want to keep this JWT for local test only.

**Review by AndlerRL — CHANGES_REQUESTED** [review]
Is looking very good! The toggle seems a little bit smaller from the designs (missing padding around the icons). [See here](https://www.figma.com/design/nDZnCkmXwOo2I71TXDBQTs/Masterbots-AI?node-id=1898-1563) for details. I noticed some improvements that we can do before to merge.

**Comment by sheriffjimoh** [comment]
Video : https://drive.google.com/file/d/1LYh91u2Xp2vV99UCQ7KLv1F5cnqUUI6M/view?usp=sharing
@AndlerRL

**Comment by AndlerRL** [comment]
@sheriffjimoh -- can you check the conflicts? PR looks good to go.

**Review by AndlerRL — APPROVED** [review]

**Comment by AndlerRL** [comment]
> @sheriffjimoh -- can you check the conflicts? PR looks good to go.

I fixed the conflicts issues on the PR. Merging!

**Pull request merged**